TerrariaServerAPI/TerrariaApi.Server/ServerApi.cs

```diff
@@ -225,6 +227,30 @@ internal static void HandleCommandLine(string[] parms)
 					case "-autocreate":
 						{
 							game.autoCreate(arg.Value);
+							isAutoCreating = true;
+
+							break;
+						}
+					case "-difficulty":
+						{
+							if (!isAutoCreating)
+							{
+								// If we're not creating a world, ignore this flag
```

Comment:
```suggestion
								LogWriter.ServerWriteLine("Ignoring difficulty command line flag because server is starting in interactive mode without autocreate", TraceLevel.Warning);
```